test/conftest.py

```diff
@@ -328,37 +328,38 @@ def t_sfdr(classname, devicename, channel, param_set, sfdr_min):
     # Set custom device parameters
     for p in param_set.keys():
         setattr(sdr, p, param_set[p])
-    time.sleep(5)  # Wait for QEC to kick in
+    # time.sleep(5)  # Wait for QEC to kick in
     # Set common buffer settings
-    N = 2 ** 14
     sdr.tx_cyclic_buffer = True
-    sdr.tx_enabled_channels = [channel]
-    sdr.tx_buffer_size = N * 2 * len(sdr.tx_enabled_channels)
-    sdr.rx_enabled_channels = [channel]
-    sdr.rx_buffer_size = N * 2 * len(sdr.rx_enabled_channels)
+    sdr.rx_buffer_size = 2 ** 15
+    if sdr._num_tx_channels > 2:
+        sdr.tx_enabled_channels = [channel]
+    if sdr._num_rx_channels > 2:
+        sdr.rx_enabled_channels = [channel]
     # Create a sinewave waveform
     if hasattr(sdr, "sample_rate"):
         RXFS = int(sdr.sample_rate)
     else:
         RXFS = int(sdr.rx_sample_rate)
     fc = RXFS * 0.1
-
+    N = 2 ** 15
     ts = 1 / float(RXFS)
     t = np.arange(0, N * ts, ts)
-    i = np.cos(2 * np.pi * t * fc) * 2 ** 15 * 0.9
-    q = np.sin(2 * np.pi * t * fc) * 2 ** 15 * 0.9
+    i = np.cos(2 * np.pi * t * fc) * 2 ** 15 * 0.5
+    q = np.sin(2 * np.pi * t * fc) * 2 ** 15 * 0.5
     iq = i + 1j * q
     # Pass through SDR
+    window = np.hanning(sdr.rx_buffer_size)
```

Comment:
Can we move this into the spectral estimation function?
For example enable it here: https://github.com/analogdevicesinc/pyadi-iio/blob/master/test/rf/spec.py#L26